torchvision/transforms/v2/functional/_meta.py

```diff
@@ -352,14 +352,122 @@ def _clamp_bounding_boxes(
     return out_boxes.to(in_dtype)
 
 
+def _order_bounding_boxes_points(
+    bounding_boxes: torch.Tensor, indices: torch.Tensor | None = None
+) -> tuple[torch.Tensor, torch.Tensor]:
+    """Re-order points in bounding boxes based on specific criteria or provided indices.
+
+    This function reorders the points of bounding boxes either according to provided indices or
+    by a default ordering strategy. In the default strategy, (x1, y1) corresponds to the point
+    with the lowest x value. If multiple points have the same lowest x value, the point with the
+    lowest y value is chosen.
+
+    Args:
+        bounding_boxes (torch.Tensor): A tensor containing bounding box coordinates in format [x1, y1, x2, y2, x3, y3, x4, y4].
+        indices (torch.Tensor | None): Optional tensor containing indices for reordering. If None, default ordering is applied.
+
+    Returns:
+        tuple[torch.Tensor, torch.Tensor]: A tuple containing:
+            - indices: The indices used for reordering
+            - reordered_boxes: The bounding boxes with reordered points
+    """
+    if indices is None:
+        output_xyxyxyxy = bounding_boxes.clone().reshape(-1, 8)
+        x, y = output_xyxyxyxy[..., 0::2], output_xyxyxyxy[..., 1::2]
+        y_max = torch.max(y, dim=1, keepdim=True)[0]
+        _, x1 = (y_max - y).div(y_max).add(x.add(1).mul(100)).min(dim=1)
```

Comment:
Nit: we probably don't need to call `clone()` here, since we're never doing any inplace operation?

Also re this line:

```
        _, x1 = (y_max - y).div(y_max).add(x.add(1).mul(100)).min(dim=1)
```

I assume you used the `.div()`, `.add()` etc. methods because we're using them in other ops (for minor perf gains). But here since we're not using the inplace operators, it may be clearer to use normal `+`, `/` operators? Up to you